impl/src/util.rs

```diff
@@ -107,81 +109,46 @@ pub fn is_bazel_supported_platform(target: &str) -> (bool, bool) {
 ///
 /// Note, the Bazel triples must be defined in:
 /// https://github.com/bazelbuild/rules_rust/blob/master/rust/platform/platform.bzl
-pub fn get_matching_bazel_triples(target: &str) -> Result<Vec<String>> {
-  let target_exp = match target.starts_with("cfg(") {
-    true => target.to_owned(),
-    false => format!("cfg(target = \"{}\")", target),
-  };
-
-  let expression = Expression::parse(&target_exp)?;
-  let triples: Vec<String> = SUPPORTED_PLATFORM_TRIPLES
+pub fn get_matching_bazel_triples<'a>(
+  target: &str,
+  whitelist: &'a Option<HashSet<String>>,
+) -> Result<impl Iterator<Item = &'static str> + 'a> {
+  let expression = match target.starts_with("cfg(") {
+    true => Expression::parse(target),
+    false => Expression::parse(&format!("cfg(target = \"{}\")", target)),
+  }?;
+
+  let triples = SUPPORTED_PLATFORM_TRIPLES
     .iter()
-    .filter_map(|triple| {
+    .filter_map(move |triple| {
```

Comment:
Its to let rust know that `expression` is only needed used in the closure when we are filtering the targets, as such it needs to exist for the length of the iterator as bound by `'static` and `'a`. That is the `filter_map` takes ownership of `expression` and will not share it.

This is similar for `allowlist` which is given to the `filter` closure. Since we are not sharing these references we can move them where they are needed. It is important to note that this is not an optimisation but is actually the ownership rules.